component/src/test/java/io/siddhi/extension/execution/string/CharFrequencyExtensionTestCase.java

```diff
@@ -0,0 +1,182 @@
+        AssertJUnit.assertTrue(eventArrived);
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void testCharFrequencyFunctionExtensionVariableIndex() throws InterruptedException {
+        LOGGER.info("CharFrequencyFunctionExtension Test Case for Variable Index scenario");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition =
+                "define stream inputStream (symbol string, price long, volume long, character string);";
+        String query = (
+                "@info(name = 'query1') " +
+                "from inputStream select symbol , str:charFrequency(symbol,character) as charFrequency, character " +
+                "insert into outputStream;");
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(inStreamDefinition + query);
+
+        siddhiAppRuntime.addCallback("query1", new QueryCallback() {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+                for (Event event : inEvents) {
+                    count.incrementAndGet();
+                    if (count.get() == 1) {
+                        AssertJUnit.assertEquals(0L, event.getData(1));
+                        eventArrived = true;
+                    }
+                    if (count.get() == 2) {
+                        AssertJUnit.assertEquals(0L, event.getData(1));
+                        eventArrived = true;
+                    }
+                    if (count.get() == 3) {
+                        AssertJUnit.assertEquals(1L, event.getData(1));
+                        eventArrived = true;
+                    }
+                }
+
+            }
+        });
+
+        InputHandler inputHandler = siddhiAppRuntime.getInputHandler("inputStream");
+        siddhiAppRuntime.start();
+        inputHandler.send(new Object[]{"IBM", 700f, 100L, "z"});
+        inputHandler.send(new Object[]{"WSO2", 60.5f, 200L, "s"});
+        inputHandler.send(new Object[]{"XYZ", 60.5f, 200L, "Y"});
+        SiddhiTestHelper.waitForEvents(100, 3, count, 60000);
+        AssertJUnit.assertEquals(3, count.get());
+        AssertJUnit.assertTrue(eventArrived);
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test(enabled = false)
+    public void testCharFrequencyFunctionExtensionInvalidChar() throws InterruptedException {
+        LOGGER.info("CharFrequencyFunctionExtension TestCase for InvalidChar case");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String inStreamDefinition =
+                "@OnError(action='STREAM')" +
+                "define stream inputStream (symbol string, price long, volume long);";
+        String query = (
+                "@info(name = 'query1') " +
+                "from inputStream " +
+                "select symbol, str:charFrequency(symbol, '33') as charFrequency " +
```

Comment:
What is this test case for? will this check for string "33"

Reply:
This is failure test case to test if second parameter contains more than one character. I tried to use error streams to get the dropped event and assert